backend/cmm_helpers.ml

```diff
@@ -793,6 +792,15 @@ let addr_array_set arr ofs newval dbg =
                  ty_args = []},
       [array_indexing log2_size_addr arr ofs dbg; newval], dbg)
 
+let addr_array_set_local arr ofs newval dbg =
+  Cop(Cextcall { func = "caml_modify_local"; ty = typ_void; alloc = false;
+                 builtin = false;
+                 returns = true;
+                 effects = Arbitrary_effects;
+                 coeffects = Has_coeffects;
+                 ty_args = []},
+      [arr; untag_int ofs dbg; newval], dbg)
```

Comment:
Does it need `array_indexing log2_size_addr` here, like other `addr_array_*` functions?

Reply:
It does not.  There was test coverage for this because I got it wrong the first time.